packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart

```diff
@@ -102,6 +102,41 @@ abstract class VideoPlayerPlatform extends PlatformInterface {
   Future<void> setMixWithOthers(bool mixWithOthers) {
     throw UnimplementedError('setMixWithOthers() has not been implemented.');
   }
+
+  /// Returns true if picture-in-picture is supported on the device.
+  Future<bool> isPictureInPictureSupported() async => false;
+
+  /// Enable/disable to start picture-in-picture automatically when the app goes to the background.
+  Future<void> setAutomaticallyStartsPictureInPicture({
+    required int textureId,
+    required bool enableStartPictureInPictureAutomaticallyFromInline,
+  }) {
+    throw UnimplementedError(
+        'setAutomaticallyStartsPictureInPicture() has not been implemented.');
+  }
+
+  /// Set the location of the video player view. So picture-in-picture can use it for animating.
+  /// The rect will represent the location of the video player view in Flutter. The rect will be
+  /// passed to the platform to position the native picture-in-picture overlay correctly.
+  Future<void> setPictureInPictureOverlayRect({
```

Comment:
We should future-proof this against other platforms potentially needing other information, by making it `setPictureInPictureOverlaySettings`, and replacing the `Rect` with a new `PictureInPictureOverlaySettings` (`@immutable`) class that currently contains just a `Rect`. Then if we need to add more things later we can do so without a breaking change.

Reply:
Fair point. I will do it like this:

```
class PictureInPictureOverlaySettings {
  PictureInPictureOverlaySettings({
    required this.ios,
  });

  PictureInPictureOverlaySettings ios;
}

class PictureInPictureOverlayIOSSettings {
  PictureInPictureOverlayIOSSettings({
    required this.rect,
  });

  PictureInPictureOverlayRect rect;
}

class PictureInPictureOverlayRect {
  PictureInPictureOverlayRect({
    required this.top,
    required this.left,
    required this.width,
    required this.height,
  });

  double top;
  double left;
  double width;
  double height;
}
```

That way we can add, remove update fields to each platform without an impact on the other platform

Reply:
Platform-specific settings need to be done via inheritance, not composition, since composition doesn't really support federation. `webview_flutter` has examples of that pattern.

In this case, I don't know that we need to do that yet though; starting with a generic settings object with a rect is fine, and if there are platform-specific things later we can add subclasses.

Reply:
Alright, I thought I saw some implementation like that before, must have been a non flutter.dev package. I will update that

Reply:
That doesn't seem so inherently iOS-specific that it needs a subclass; other platforms could quite plausibly need the same information, and those that don't can just ignore it.

Reply:
Alright, thanks already, really helpful feedback